src/content/basics/cluster-size-autoscaling/index.md

```diff
@@ -0,0 +1,66 @@
++++
+title = "Cluster Size and Autoscaling"
+description = "This article explains options you have for defining the size of a Kubernetes cluster with Giant Swarm, and automatically scaling it"
+date = "2019-02-01"
+weight = 45
+type = "page"
+categories = ["basics"]
++++
+
+# Cluster Size and Autoscaling
+
+Starting with release version {{% first_aws_autoscaling_version %}} for AWS, you can leverage the benefits of the [Kubernetes autoscaler](https://github.com/kubernetes/autoscaler) to define the number of worker nodes in a cluster based on demand.
+The autoscaler is provided with every cluster of version >= {{% first_aws_autoscaling_version %}} on AWS.
+
+On Giant Swarm installations on Azure, on bare-metal, and on AWS prior to version {{% first_aws_autoscaling_version %}}, the cluster size would be defined statically.
+
+## Setting scaling limits
+
+With the autoscaler taking control over the number of worker nodes, you only define the bounds or limits in which the cluster size can vary. This is possible both during cluster creation as well as after creation.
+
+To enforce an exact cluster size and **effectively disable the autoscaler**, simply set the minimum and maximum worker node count to the same value. This is also the recommended way in case you want to control the number of worker nodes through some external tooling via the Giant Swarm API.
+
+## Minimal and default cluster size
+
+When creating a cluster without specifying the number of worker nodes, {{% default_cluster_size_worker_nodes %}} worker nodes will be created. This is also the minimal number supported by Giant Swarm. On AWS starting with release version {{% first_aws_autoscaling_version %}}, when not specified, the maximum number of worker nodes is also set to {{% default_cluster_size_worker_nodes %}}.
+
+Technically, while you may be able to create and run smaller clusters successfully, we don't encourage this due to reduced resilience. We explicitly deactivate all sorts of alerts for clusters with less than {{% minimal_supported_cluster_size_worker_nodes %}} worker nodes and won't get notified in case of any problems.
+
+## Minimal worker node instance type
+
+Relying on autoscaling may have an effect on the instance type you should use for your cluster.
+To illustrate this more detail, it is helpful to explain how the autoscaler determines whether a node is needed or not.
+
+The autoscaler periodically calculates the utilization of a node based on CPU and memory requests of all running pods, compared to the node's total capacity.
+
+**Note:** this means that your pods need sensible CPU and memory requests in order to inform the autoscaler about the actual node utilization.
```

Comment:
Does it mean that the user **must** set up requests (and perhaps limits) property in the YAML configuration? It is not clear here if any action is expected from the user.

Reply:
Technically it's not mandatory, but scheduling doesn't work as expected and given the workloads we have experienced, in practice it is mandatory. Nevertheless, there's no validation nor similar mechanisms preventing from submitting ill-configured PODs. Hence this wording, I think.

Reply:
I see but we should probably enforce this by writing that. It's true that no validation for those kind of missing things (limits and requests) but maybe it worths to communicate this. we take two pigeons with one bean:

- customers will enforce requests and limits in their applications (which is generally a very good practice)
- cluster autoscaler will work smooth and properly

cc @marians

Reply:
I will try to make it clearer.

Reply:
Change note sentence to `This means that your pods MUST be configured with CPU and memory requests in order to inform the autoscaler about the actual node utilization.`. How about that?

Reply:
Love that.